src/subSystems.ts

```diff
@@ -0,0 +1,137 @@
+const log = require('@kth/log')
+log.init()
+
+import type { MonitoredSystem, ProbeType, SystemCheckResult } from './types'
+
+const SYSTEM_CHECK_TIMEOUT = 5000
+
+export const filterSystems = (probeType: ProbeType, systems: MonitoredSystem[] = []) => {
+  if (probeType === 'full') {
+    return systems
+  }
+
+  if (probeType === 'readyness') {
+    return systems.filter(system => ['redis', 'mongodb', 'sqldb'].includes(system.key))
+  }
+
+  return []
+}
+
+export const checkSystems = async (systems: MonitoredSystem[]) => Promise.all(systems.map(checkSystemOrTimeout))
+
+const checkSystemOrTimeout = (system: MonitoredSystem) =>
+  Promise.any([
+    // Return successfull result or timeout, depenting on what resolves fastest
+    checkSystem(system),
+    timeoutSystem(system),
+  ])
+
+const timeoutSystem = async (system: MonitoredSystem): Promise<MonitoredSystem> => {
+  const result = { ...system }
+
+  await sleep(SYSTEM_CHECK_TIMEOUT)
+
+  result.result = { status: false, message: 'system timed out' }
+
+  return result
+}
+
+const checkSystem = async (system: MonitoredSystem): Promise<MonitoredSystem> => {
+  const result = { ...system }
+
+  if (isMongodbSystem(system)) {
+    result.result = await checkMongodbSystem(system)
+  } else if (isRedisSystem(system)) {
+    result.result = await checkRedisSystem(system)
+  } else if (isKthApiSystem(system)) {
+    result.result = await checkKthApiSystem(system)
+  } else if (isSqldbSystem(system)) {
+    result.result = await checkSqldbSystem(system)
+  } else {
+    result.ignored = true
+    log.warn('@kth/monitor - Unknown system', system)
+  }
+  return result
+}
+
+const isMongodbSystem = (system: MonitoredSystem) => system.key === 'mongodb'
+const isRedisSystem = (system: MonitoredSystem) => system.key === 'redis'
+const isKthApiSystem = (system: MonitoredSystem) => system.endpoint != undefined
+const isSqldbSystem = (system: MonitoredSystem) => system.key === 'sqldb'
+
+const checkMongodbSystem = async (system: MonitoredSystem): Promise<SystemCheckResult> => {
+  if (typeof system.db?.isOk != 'function') {
+    return { status: false, message: 'invalid configuration' }
+  }
+
+  if ((await system.db?.isOk()) === true) {
+    return { status: true }
+  }
+
+  return { status: false }
+}
+
+const checkRedisSystem = async (system: MonitoredSystem): Promise<SystemCheckResult> => {
```

Comment:
You could also cast the return value instead of having `Promise<SystemCheckResult>` in the function head, but I see no direct advantage of that in our case so feel free to keep it like this